Alicerce/Sources/Extensions/Collection.swift

```diff
@@ -0,0 +1,51 @@
+//
+//  Collection.swift
+//  Alicerce
+//
+//  Created by Luís Afonso on 18/12/2016.
+//  Copyright © 2016 Mindera. All rights reserved.
+//
+
+import Foundation
+
+// MARK: - Array
+
+func -<Key: Hashable, Value: AnyObject>(left: [[Key : Value]], right: [[Key : Value]]) -> [[Key : Value]] {
+    var final = left
+    
+    for (index, element) in right.enumerated() {
+        final[index] = (left[index] - element)
+    }
+    
+    return final
+}
+
+// MARK: - Dictionary
+
+func -<Key: Hashable, Value: AnyObject>(left: [Key : Value], right: [Key : Value]) -> [Key : Value] {
+    var final = left
+    
+    for (key, value) in right {
+        if let val = left[key], val === value {
+            final[key] = nil
+        }
+    }
+    
+    return final
+}
+
+func +<Key, Value>(left: [Key : Value], right: [Key : Value]) -> [Key : Value] {
+    var final = left
+    
+    for (key, value) in right {
+        final[key] = value
```

Comment:
This behavior might not be obvious, because we overwrite existing values on `left` with the ones on `right` when `key` matches. Don't know if adding a "non destructive" version would be useful though 🤔